src/middlewared/middlewared/plugins/smb.py

```diff
@@ -504,11 +504,8 @@ async def netif_wait(self, timeout=120):
         """
         found_sentinel = False
         while timeout >= 0 and not found_sentinel:
-            try:
-                os.remove(NETIF_COMPLETE_SENTINEL)
+            if os.path.exists(NETIF_COMPLETE_SENTINEL):
```

Comment:
We shouldn't be doing file IO in async coroutine. You can either make method synchronous or run this in thread via self.middleware.run_in_thread.